app/models/workers_scaling/experiment_resources_interface.rb

```diff
@@ -9,90 +9,97 @@ class ExperimentResourcesInterface
     MAXIMUM_NUMBER_OF_FAILED_WORKERS = 3
 
     ##
-    # Params:
-    # * experiment
-    # * user_id
-    # * allowed_infrastructures
-    #     Variable storing user-defined workers limits for each infrastructure configuration
-    #     Only infrastructures specified here can be used by experiment.
-    #     Format: [{infrastructure: <infrastructure_configuration>, limit: <limit>}, ...]
-    #     <infrastructure_configuration> - hash with infrastructure configuration
-    #     <limit> - integer
-    def initialize(experiment, user_id, allowed_infrastructures)
+    # @param experiment [Experiment]
+    # @param user_id [BSON::ObjectId, String]
+    # @param allowed_resource_configurations [Array<Hash, ActiveSupport::HashWithIndifferentAccess>]
+    #   Variable storing user-defined workers limits for each resource configuration
+    #   Only resource configuration specified here can be used by experiment.
+    #   Format: [{resource_configuration: <resource_configuration>, limit: <limit>}, ...]
+    #     <resource_configuration> - hash with resource configuration
+    #     <limit> - Fixnum
+    def initialize(experiment, user_id, allowed_resource_configurations)
       @experiment = experiment
       @user_id = BSON::ObjectId(user_id.to_s)
       @facades_cache = {}
-      @allowed_infrastructures = allowed_infrastructures.map {|x| ActiveSupport::HashWithIndifferentAccess.new(x)}
+      @allowed_resource_configurations = allowed_resource_configurations.map do |x|
+        ActiveSupport::HashWithIndifferentAccess.new(x)
+      end
     end
 
     ##
-    # Returns list of enabled infrastructure configurations for experiment
-    # Infrastructure configuration format: {name: <name>, params: {<params>}}
-    def get_enabled_infrastructures
+    # Returns list of enabled resource configurations for experiment
```

Comment:
To klasa ExperimentResourcesInterface - jest przypisana do ekperymentu, otrzymuje go w konstruktorze.